src/NuGet.Core/NuGet.LibraryModel/LibraryDependencyTargetUtils.cs

```diff
@@ -14,35 +14,19 @@ public class LibraryDependencyTargetUtils
         /// <summary>
         /// Convert flag string into a LibraryTypeFlag.
         /// </summary>
-        public static LibraryDependencyTarget Parse(string flag)
+        public static LibraryDependencyTarget Parse(string flags)
         {
             // If the LibraryDependency does not have a flag value it is considered all
-            if (string.IsNullOrEmpty(flag))
+            if (string.IsNullOrEmpty(flags))
             {
                 return LibraryDependencyTarget.All;
             }
 
-            switch (flag.ToLowerInvariant())
```

Comment:
This list restricts the allowed input values, there are some flags which project.json should not allow such as PackageProjectExternal